doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -4,99 +4,82 @@
 Programmable proof search
 =========================
 
-.. tacn:: auto
-   :name: auto
+.. tacn:: auto {? @int_or_var } {? @auto_using } {? @hintbases }
 
-   This tactic implements a Prolog-like resolution procedure to solve the
-   current goal. It first tries to solve the goal using the :tacn:`assumption`
-   tactic, then it reduces the goal to an atomic one using :tacn:`intros` and
-   introduces the newly generated hypotheses as hints. Then it looks at
-   the list of tactics associated to the head symbol of the goal and
-   tries to apply one of them (starting from the tactics with lower
-   cost). This process is recursively applied to the generated subgoals.
```

Comment:
I think it is problematic that this description is moved so far down below. IMHO it is short enough to be kept here.